CMakeLists.txt

```diff
@@ -60,6 +60,28 @@ endif()
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC -DDROP_CGAL ${CPP_STANDARD_FLAGS} -Wall -Wextra -Wpedantic -Wno-unused-variable -Wno-unused-parameter")
 
 
+#=====================================================================
+# check if the linker supports as needed or not
+#   (needed for ROOT dict, specifically on Ubuntu)
+set(STORE_REQUIRED_FLAGS ${CMAKE_REQUIRED_FLAGS})
+include(CheckCCompilerFlag)
+set(CMAKE_REQUIRED_FLAGS "-Wl,--no-as-needed")
+check_c_compiler_flag("" TEST_NO_AS_NEEDED)
```

Comment:
the variables are added to the internal cache (https://cmake.org/cmake/help/latest/module/CheckCCompilerFlag.html) So it should be perfectly fine to include these checks inside the function body.